### PR TITLE
Add support for displaying dialogs

### PR DESCRIPTION
**Pull request by narenkmanoharan**
- [x] Add integration test

### CODE REVIEW

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  @VisibleForTesting internal var dialogIsShowing: Boolean = false
+
+  fun showDialog(dialogCreator: DialogCreator) {
+    this.dialogCreator = dialogCreator
+    this.dialogIsShowing = true
+    createDialog()
+  }
+
+  override fun create(context: Context) {
```
Comment (narenkmanoharan) [inline magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:23]: I love lifecycle aware haha

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (dialog != null) {
+      dialogIsShowing = dialog!!.isShowing
+      dialog!!.setOnDismissListener(null)
+      dialog!!.dismiss()
+      dialog = null
+    }
+  }
+}
+
+typealias DialogCreator = (Context) -> Dialog
```
Comment (ryanmoelter) [inline magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:56]: nit: might want to put this at the top of the file, feels kind of weird dangling at the bottom haha.

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  private var dialogCreator: DialogCreator? = null
+  private var dialog: Dialog? = null
+  private var context: Context? = null
+
+  @VisibleForTesting internal var dialogIsShowing: Boolean = false
+
+  fun showDialog(dialogCreator: DialogCreator) {
+    this.dialogCreator = dialogCreator
+    this.dialogIsShowing = true
```
Comment (ryanmoelter) [inline magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:19]: Should we set this to false when the dialog is dismissed?

File: magellan-journey-sample/src/androidTest/java/com/wealthfront/magellan/uitest/DialogTest.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  var activityRule = ActivityTestRule(MainActivity::class.java)
+
+  @Test
+  fun showDialog() {
+    onView(withId(R.id.nextJourney)).perform(click())
+
+    onView(withId(R.id.why)).perform(click())
+    assertShown { text("Hello") }
+    assertShown { text("Are you sure about this?") }
+    pressBack()
```
[the remark this answers is not in the document]
Comment (narenkmanoharan) [reply magellan-journey-sample/src/androidTest/java/com/wealthfront/magellan/uitest/DialogTest.kt:27]: Oops. this test fails. Removed the pressBack.

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt (hunk 2 above)
Comment (narenkmanoharan) [reply magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:56]: d

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt (hunk 3 above)
Comment (narenkmanoharan) [reply magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:19]: d

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      dialog!!.show()
+      dialogIsShowing = true
+    }
+  }
+
+  private fun destroyDialog() {
+    if (dialog != null) {
+      dialog!!.setOnDismissListener(null)
+      dialog!!.dismiss()
+      dialogIsShowing = false
```
[the remark this answers is not in the document]
Comment (narenkmanoharan) [reply magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:52]: Hmm. no. This sets the `dialogIsShowing` to false and we only show dialog if there isn't a dialog already show with this component.

[the remark this answers is not in the document]
Comment (narenkmanoharan) [reply magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt]: We will show the dialog again if it goes through config changes..

**Pull request merged**